api/task_manager.cc

```diff
@@ -179,7 +180,51 @@ void set_task_manager(http_context& ctx, routes& r) {
                 return make_foreign(task);
             });
         }));
-        co_return co_await retrieve_status(std::move(task));
+        auto s = co_await retrieve_status(task);
+        co_return make_status(s);
```

Comment:
Please extract this transition into a separate patch, before adding `get_task_status_recursively`